DXRModules/DeusEx/Classes/DXREvents.uc

```diff
@@ -1959,98 +1959,139 @@ function string RemapBingoEvent(string eventname)
 
 }
 
-static function string GetBingoFailedGoals(DXRando dxr, string eventname, out string failed2)
+static function GetBingoFailedGoals(DXRando dxr, string eventname, out string failed[5])
 {
-    failed2 = "";
+    local int i;
+
+    for (i = 0; i < ArrayCount(failed); i++) {
+        failed[i] = "";
+    }
 
     // keep in mind that a goal can only be marked as failed if it isn't already marked as completed
     switch (eventname) {
         case "JuanLebedev_Dead":
-            return "LebedevLived";
+            failed[0] = "LebedevLived";
+            return;
         case "Aimee_Dead":
         case "LeMerchant_Dead":
-            return "AimeeLeMerchantLived";
+            failed[0] = "AimeeLeMerchantLived";
+            return;
         case "MaggieChow_Dead":
-            return "MaggieLived";
+            failed[0] = "MaggieLived";
         case "FordSchick_Dead":
-            return "FordSchickRescued";
+            failed[0] = "FordSchickRescued";
+            return;
         case "AlleyBum_Dead":
-            return "AlleyBumRescued";
+            failed[0] = "AlleyBumRescued";
+            return;
         case "Camille_Dead":
-            return "CamilleConvosDone";
+            failed[0] = "CamilleConvosDone";
+            return;
         case "Miguel_Dead":
-            return "nsfwander";
+            failed[0] = "Terrorist_peeptime";
+            failed[1] = "Terrorist_ClassDead";
```

Comment:
hardcoded array indexes can be a pain, especially if you're rearranging things, or adding/removing things
```suggestion
            failed[i++] = "Terrorist_ClassDead";
```